flink-ml-lib/src/test/java/org/apache/flink/ml/util/MockKVStore.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/** Class that manages global key-value pairs used in unit tests. */
+@SuppressWarnings({"unchecked"})
+public class MockKVStore {
```

Comment:
`TestMetricReporter` is not singleton, as you can see the following in the test:
```java
config.setString(
        "metrics.reporter.test_reporter.class", TestMetricReporter.class.getName());
```
Flink will use this information to instantiate `TestMetricReporter` on its own, so it will not be singleton.

But I agree that we can remove `MockKVStore` for now as there is only `TestMetricReporter` using it. The `TestMeticReporter` can store values in its own static variables and provide the static get method to acquire them from Flink clients in test cases.

Reply:
I see. Do you think we can replace `TestMetricReporter ` with Flink's InMemoryReporter?

Reply:
Solved as described in the other comment about TestMetricReporter.